eclipse-collections/src/main/java/org/eclipse/collections/impl/string/immutable/CodePointAdapter.java

```diff
@@ -781,6 +783,12 @@ public int hashCode()
         return target.toImmutable();
     }
 
+    @Override
+    public Spliterator.OfInt primitiveSpliterator()
+    {
+        return Spliterators.spliterator(toArray(),Spliterator.ORDERED);
```

Comment:
Calling toArray() is O(n). Could we adapt the iterator instead? Is that more work than manually implementing the spliterator?

Reply:
Could you delegate to `this.adapted.codePoints()`, which is an `IntStream`?

Reply:
Yes, I had a look at that implementation and it does a good job of splitting without O(n).